spec/02-integration/05-proxy/02-router_spec.lua

```diff
@@ -2308,6 +2308,33 @@ for _, strategy in helpers.each_strategy() do
       end
     end)
 
+    it("#db worker respawn correctly rebuilds router", function()
+      local admin_client = helpers.admin_client()
+
+      local res = assert(admin_client:post("/routes", {
+        headers = { ["Content-Type"] = "application/json" },
+        body = {
+          paths = { "/foo" },
+        },
+      }))
+      assert.res_status(201, res)
+      admin_client:close()
+
+      assert(helpers.signal_workers(nil, "-TERM"))
+
+      proxy_client:close()
+      proxy_client = helpers.proxy_client()
+
+      local res = assert(proxy_client:send {
```

Comment:
Is there any risk that this request could be handled by a worker that hasn't been killed yet?

Reply:
Seems like a low risk given SIGTERM and not SIGQUIT, but `helpers.get_kong_workers` and `helpers.wait_until_no_common_workers` might be useful.

Reply:
It does not seems to be flappy according to my tests on CI. It always catches the issue and never fails due to connection reset.